File: shared/src/commonMain/kotlin/com/example/plauenblod/feature/route/model/util/PointsCalculator.kt
package com.example.plauenblod.feature.route.model.util

import com.example.plauenblod.feature.route.model.routeProperty.Difficulty
import com.example.plauenblod.feature.route.model.routeProperty.HoldColor

fun calculatePoints(difficulty: Difficulty, isFlash: Boolean = false): Int {
    val basePoints = when (difficulty) {
        Difficulty.WHITE -> 10
        Difficulty.YELLOW -> 20
        Difficulty.BLUE -> 30
        Difficulty.GREEN -> 40
        Difficulty.RED -> 50
        Difficulty.BROWN -> 60
        Difficulty.PINK -> 0
    }
    return if (isFlash && basePoints > 0) basePoints + 2 else basePoints
}

File: Bouldery/feature/route/service/PointsCalculator.swift
//
//  PointsCalculator.swift
//  Bouldery
//
//  Created by Kim Reuter on 20.08.25.
//

import Foundation

struct PointsCalculator {
    static func points(for difficulty: Difficulty, isFlash: Bool = false) -> Int {
        let base: Int
        switch difficulty {
        case .PINK:  return 0
        case .WHITE: return 10
        case .YELLOW:return 20
        case .BLUE:  return 30
        case .GREEN: return 40
        case .RED:   return 50
        case .BROWN: return 60
        }
        return (isFlash && base > 0) ? base + 10 : base
    }
}
